spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -151,36 +153,41 @@ public void destroyAll() {
 	 * Return a page-able list of {@link TaskDefinitionResource} defined tasks.
 	 *
 	 * @param pageable page-able collection of {@code TaskDefinitionResource}
-	 * @param search optional findByTaskNameContains parameter
+	 * @param taskName optional findByTaskNameContains parameter
```

Comment:
I agree, `search` is not a good name for this parameter, especially now that another filter parameter has been added. However, this will be a breaking change to the controller API. 

Let's keep the original endpoint, mark it deprecated, and have it simply call the newly added method, something like: 
```java
	@RequestMapping(value = "", method = RequestMethod.GET)
	@ResponseStatus(HttpStatus.OK)
	@Deprecated
	public PagedModel<? extends TaskDefinitionResource> list(Pageable pageable,
			@RequestParam(required = false) String taskName,
			@RequestParam(required = false) boolean manifest,
			@RequestParam(required = false) String dslText,
			PagedResourcesAssembler<TaskExecutionAwareTaskDefinition> assembler) {
		return list(pageable, taskName, null, dslText, manifest, assembler);
	}
```

The other option would be to keep a single endpoint impl method and add back in the `search` param and javadoc the param as deprecated. 

Wdyt? 

In fact, I was suprised that tests did not break because of this change. When I ran `TaskControllerTests` (that looks like the test that checks this task def. controller) locally it does fail. @corneil can you please take a look at why https://github.com/spring-cloud/spring-cloud-dataflow/actions/runs/4995230765/jobs/8946846252?pr=5344 is not running all tests? It runs some, but you will not see `Starting TaskControllerTests` in the build logs like you do when running local.

Reply:
I re-added the old method and marked it as deprecated. The old method delegates to the new one and the local tests should not fail any longer.

Reply:
I saw that two methods is not possible due to rest restrictions and ambiguous mapping. I changed the implementation to only deprecated the parameter. I think the UI should not be affected by this change as it hands over the parameters to the backend.

Reply:
Yeh, I wondered in the back of my head whether or not the method would be ambiguous (due to mappings).